src/librustc/ty/context.rs

```diff
@@ -1072,6 +1072,23 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         local as usize == global as usize
     }
 
+    /// Returns true if this function must conform to `min_const_fn`
+    pub fn is_min_const_fn(self, def_id: DefId) -> bool {
+        match self.lookup_stability(def_id) {
+            // stable functions with unstable const fn don't need to conform
```

Comment:
I thought we got rid of `rustc_const_unstable` or does this do something else?

Reply:
This is indeed about `#[rustc_const_unstable]`, I'll add some tests to ensure that you can't call one of these from a `min_const_fn` (so any `const fn` without `#[rustc_const_unstable]` in libstd or any `const fn` where only the `min_const_fn` feature gate is active.

We need to keep this, but it's orthogonal to `min_const_fn`. We have three kinds const fn now:

* `min_const_fn` (can only call other `min_const_fn`)
    * this is what any stable const fn in the libstd is
* `#[unstable]` noone can use this without a feature gate
* `#[rustc_const_unstable]` noone can use this without the specific "const" feature gate

Reply:
Sounds good 👍